src/components/Widgets/ListControl.js

```diff
@@ -102,8 +102,8 @@ export default class ListControl extends Component {
     return (e) => {
       e.preventDefault();
       const { value, metadata, onChange, forID } = this.props;
-      const parsedMetadata = { [forID]: metadata.removeIn(value.get(index).valueSeq()) };
-      onChange(value.remove(index), parsedMetadata);
+      const parsedMetadata = metadata && { [forID]: metadata.removeIn(value.get(index).valueSeq()) };
+      onChange(value.remove(index), parsedMetadata || undefined);
```

Comment:
Our lack of testing is biting us so much that I'm pretty strongly against changing contracts anywhere unnecessarily. That said, now that I look at it, `parsedMetadata` would actually be undefined in this case, so we can definitely remove it.